cmd/kubeadm/app/util/config/initconfiguration.go

```diff
@@ -85,9 +86,12 @@ func SetBootstrapTokensDynamicDefaults(cfg *[]kubeadmapi.BootstrapToken) error {
 // SetNodeRegistrationDynamicDefaults checks and sets configuration values for the NodeRegistration object
 func SetNodeRegistrationDynamicDefaults(cfg *kubeadmapi.NodeRegistrationOptions, ControlPlaneTaint bool) error {
 	var err error
-	cfg.Name, err = nodeutil.GetHostname(cfg.Name)
-	if err != nil {
-		return err
+	cfg.Name = strings.TrimSpace(cfg.Name)
```

Comment:
aren't we supposed to trim after?

Reply:
the trimming logic seems to solve a weird problem that i don't understand.
why didn't you duplicate the whole `func GetHostname(hostnameOverride string) (string, error) {` under kubeadm?

Reply:
@neolit123 

> why didn't you duplicate the whole func GetHostname(hostnameOverride string) 

I did it some time ago https://github.com/kubernetes/kubernetes/pull/79477 Feel free to accept that PR if you want.

Reply:
> a weird problem that i don't understand.

That was caused by not using ToLower. Should be fixed now.